src/mono/mono/metadata/sgen-bridge.h

```diff
@@ -103,7 +103,7 @@ typedef struct {
  */
 MONO_API void mono_gc_register_bridge_callbacks (MonoGCBridgeCallbacks *callbacks);
 
-MONO_API void mono_gc_wait_for_bridge_processing (void);
+MONO_API MONO_RT_EXTERNAL_ONLY void mono_gc_wait_for_bridge_processing (void);
```

Comment:
I assume this MONO_RT_EXTERNAL_ONLY thing is to mark it as something that should only be called externally, not by the runtime itself? Is there some static checking that prevents such calls?

Reply:
It expands to an `__attribute__((deprecated))` when it's compiled inside the runtime, so we get a build error.

the general pattern we've been following with GC Unsafe transitions in MONO_API functions is that we add an `_internal` vesion which _doesn't_ do the transition (because inside the runtime we're in GC Unsafe mode most of the time and do transitions explicitly), and an external-only API function that performs the transition explicitly because they are primarily called from outside the runtime which is mostly in GC Safe mode.  (transitions to enter unsafe are ok to nest, so it would be harmless to dispense with the `_internal` function, but it's a bit nicer to avoid the extraneous function call)